packages/transaction-controller/src/utils/batch.ts

```diff
@@ -410,14 +412,25 @@ async function addTransactionBatchWithHook(
   const collectHook = new CollectPublishHook(transactionCount);
   try {
     if (requireApproval && useHook) {
+      const { gasLimit, transactions: transactionsWithGas } =
+        await simulateGasBatch({
+          chainId,
+          from,
+          transactions: nestedTransactions,
+        });
+
+      // resigned the transactions with simulated gas
+      nestedTransactions = transactionsWithGas;
```

Comment:
Is this necessary? The individual gas limits aren't needed yet in the UI, and it would greatly simplify the logic if we can just return a single gas limit to set?

Reply:
Do you mean not passing it now here to `nestedTransactions` , but keep the logic in `simulateGasBatch` so we can useit  in the future?
I removed the reassignment, but still kept the logic in `simulateGasBatch`.